packages/validator/src/services/indices.ts

```diff
@@ -85,9 +85,13 @@ export class IndicesService {
 
     this.pollValidatorIndicesPromise = this.pollValidatorIndicesInternal(pubkeysHex);
     // Once the pollValidatorIndicesInternal() resolves or rejects null the cached promise so it can be called again.
-    this.pollValidatorIndicesPromise.finally(() => {
-      this.pollValidatorIndicesPromise = null;
-    });
+    this.pollValidatorIndicesPromise
+      .catch((err) => {
+        this.logger.error("Error polling validator indices", {}, err);
+      })
+      .finally(() => {
+        this.pollValidatorIndicesPromise = null;
+      });
```

Comment:
@nazarhussain I think we should revert this change as it changes the behavior of this function. The returned promise will never reject now. This makes it impossible to handle the error upstream, e.g. in `pollBuilderValidatorRegistration` which logs the error with much more context.

https://github.com/ChainSafe/lodestar/blob/a11427edba4a0cabcb4b212a9eeef78199bea246/packages/validator/src/services/prepareBeaconProposer.ts#L83-L85

In general I think we should avoid introducing behavior changes in PRs that update dependencies or fix linting issues.

Reply:
Yes I will open a PR shortly to revert that line. 

The change was introduced because the code was causing the limiting error "floating promise" with updated eslint.

Reply:
> code was causing the limiting error "floating promise" with updated eslint

This sounds like a bug in the linter, if the promise is returned it is the job of the caller to handle the promise.

Reply:
No it's not a bug in linter. if the promise is returned in the chain then it's not a floating promise. If promise is attached a handler independently of the return statement then linter can't detect if it was ever returned.

Reply:
> If promise is attached a handler independently of the return statement then linter can't detect if it was ever returned.

Still doesn't seem right to me, the linter should detect that as it is returned either way but can just ignore the lint rule in that case as you did in the PR